tools/pythonpkg/duckdb/experimental/spark/sql/dataframe.py

```diff
@@ -846,9 +846,23 @@ def unionByName(
         |NULL|   4|   5|   6|
         +----+----+----+----+
         """
-        if not allowMissingColumns:
-            raise ContributionsAcceptedError
-        raise NotImplementedError
+        if allowMissingColumns:
+            for col in set(self.columns) - set(other.columns):
```

Comment:
I also wonder if this has side effects on the passed in `other`, I suspect it does
It shouldn't.